fast64_internal/mk64/f3d_course_parser.py

```diff
@@ -0,0 +1,338 @@
+import re
+import traceback
+from mathutils import Vector
+from ..f3d.f3d_gbi import F3D
+from ..f3d.f3d_material import RDPSettings
+from ..f3d.f3d_parser import F3DParsedCommands, ParsedMacro, math_eval, parseDLData, parseVertexData
+from ..f3d.f3d_writer import F3DVert
+from ..utility import PluginError, float_from_u16_str, gammaInverseValue, int_from_s16_str, readFile, unpackNormal
+
+
+def courseVertexFormatPatterns():
+    # position, uv, color/normal
+    return (
+        # decomp format
+        r"\{\s*"
+        r"\{+([^,\}]*),([^,\}]*),([^,\}]*)\}\s*,\s*"
+        r"\{([^,\}]*),([^,\}]*)\}\s*,\s*"
+        r"\{MACRO_COLOR_FLAG\(([^,\}]*),([^,\}]*),([^,\}]*),([^,\}])*\),([^,\}]*)\}\s*"
+        r"\}"
+    )
+
+
+def parseCourseVtx(path: str, f3d):
+    data = readFile(path)
+    pattern = courseVertexFormatPatterns()
+    vertexData = []
+    for values in re.findall(pattern, data, re.DOTALL):
+        values = [math_eval(g, f3d) for g in values]
+        vertexData.append(
+            F3DVert(
+                Vector(values[0:3]),
+                Vector(values[3:5]),
+                Vector(values[5:8]),
+                unpackNormal(values[8]),
+                values[9],
+            )
+        )
+    return vertexData
+
+
+class MK64F3DContext:
```

Comment:
```suggestion
class MK64F3DContext(F3DContext):
```
and don't need to have `processCommands` in this class
and change the base class `F3DContext` by moving base `getVertexDataStart` into it as a method and having `processCommands` call that method

Reply:
processCommands are by default a function of F3DContext I just use a trick to avoid creating the subclass to use it